datafusion/core/src/physical_plan/sorts/index.rs

```diff
@@ -0,0 +1,42 @@
+/// A `RowIndex` identifies a specific row in a logical stream.
+///
+/// Each stream is identified by an `stream_idx` and is formed from a
```

Comment:
This code is moved into a new module, and I added some comments / ascii art to help my future self not have to read the code the next time